sky/clouds/azure.py

```diff
@@ -258,17 +258,24 @@ def get_feasible_launchable_resources(self, resources):
         def _make(instance_list):
             resource_list = []
             for instance_type in instance_list:
-                if Azure.check_disk_tier(instance_type, resources.disk_tier)[0]:
-                    r = resources.copy(
-                        cloud=Azure(),
-                        instance_type=instance_type,
-                        # Setting this to None as Azure doesn't separately bill /
-                        # attach the accelerators.  Billed as part of the VM type.
-                        accelerators=None,
-                        cpus=None,
-                        memory=None,
-                    )
-                    resource_list.append(r)
+                disk_tier = resources.disk_tier
+                ok, _ = Azure.check_disk_tier(
+                    instance_type, disk_tier or clouds.Cloud._DEFAULT_DISK_TIER)
+                if not ok and resources.disk_tier is None:
+                    # Auto failover to low disk tier when disk tier
+                    # are not specified
+                    disk_tier = 'low'
```

Comment:
Should we add a warning message `logger.warning` here to notify the user that the disk type has been downgraded?

Reply:
This could be a little bit tricky. IIUC it might not be a good idea to add a `logger.warning` here since a user might simply specify an accelerator (by `Resources(accelerators='T4')`) and get a bunch of warnings for each instance type that support that accelerator and not support medium disk tier (Premium LRS).

Reply:
I'm thinking maybe we could move this to __init__ of Resources, in which we check if the disk tier is feasible and failover to the lower tier?

Reply:
That makes sense! I am wondering if we can print the warning only when the user explicitly specifies the `disk_type` for the resources.
If that is too hard, I am ok with the current version.

Reply:
Currently, if a user explicitly specifies the disk tier, it will raise a ValueError said the specified disk tier is not supported. Shall we failover it as well?

Reply:
Ahh, that’s right. I missed it. I think the current way should be good.